base/dsfs/dataset_test.go

```diff
@@ -385,74 +387,292 @@ func TestWriteDataset(t *testing.T) {
 
 func TestGenerateCommitMessage(t *testing.T) {
 	cases := []struct {
-		ds, prev *dataset.Dataset
-		force    bool
-		expected string
-		errMsg   string
+		description string
+		prev, ds    *dataset.Dataset
+		force       bool
+		expectShort string
+		expectLong  string
+		errMsg      string
```

Comment:
please split table tests into two groups, one set of `good` cases and one set of `bad` cases:

```go
good := []struct{
  description string
  prev, ds *dataset.Dataset
  force bool
  expectShort string
  expectLong string
}{ }

bad := []struct{
  prev, ds *dataset.Dataset
  force bool
  errMsg string
}{ }
```

bad cases get to drop the description. But these tests are great!